### PR TITLE
added snapshot tests for d3 components

### PR DESCRIPTION
**Pull request by soohyunkim**
This is based off of https://github.com/tl-its-umich-edu/my-learning-analytics/pull/434 and works towards closing https://github.com/tl-its-umich-edu/my-learning-analytics/issues/426. Note that not all of the d3 components have been migrated yet so those tests will soon be added. 

D3 snapshot tests will just fail if the given data is invalid. Will leave off boundary check testing for a separate PR.

### CODE REVIEW

**Comment by Adonais0** [comment]
I met the error message shown as below when trying to run with `docker exec -it webpack_watcher npm test`:

` FAIL  assets/src/__tests__/util/math.test.js
  ● Test suite failed to run

    ReferenceError: [BABEL] /usr/src/app/assets/src/__tests__/util/math.test.js: Unknown option: /usr/src/app/node_modules/babel-preset-react-app/index.js.overrides. Check out http://babeljs.io/docs/usage/options/ for more information about options.

    A common cause of this error is the presence of a configuration options object without the corresponding preset name. Example:

    Invalid:
      `{ presets: [{option: value}] }`
    Valid:
      `{ presets: [['presetName', {option: value}]] }`

    For more detailed information on preset configuration, please see https://babeljs.io/docs/en/plugins#pluginpresets-options. (While processing preset: "/usr/src/app/node_modules/babel-preset-react-app/index.js")

      at Logger.error (node_modules/babel-core/lib/transformation/file/logger.js:41:11)
      at OptionManager.mergeOptions (node_modules/babel-core/lib/transformation/file/options/option-manager.js:226:20)
      at node_modules/babel-core/lib/transformation/file/options/option-manager.js:265:14
      at node_modules/babel-core/lib/transformation/file/options/option-manager.js:323:22
          at Array.map (<anonymous>)
      at OptionManager.resolvePresets (node_modules/babel-core/lib/transformation/file/options/option-manager.js:275:20)
      at OptionManager.mergePresets (node_modules/babel-core/lib/transformation/file/options/option-manager.js:264:10)
      at OptionManager.mergeOptions (node_modules/babel-core/lib/transformation/file/options/option-manager.js:249:14)
      at OptionManager.init (node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
      at File.initOptions (node_modules/babel-core/lib/transformation/file/index.js:212:65)`

**Comment by justin0022** [comment]
I'm getting the same error as well. I managed to fix it by importing another Babel dependency, but for some reason now Soo's tests won't pass on her machine. We're still looking into this, hope to find a solution next week.

File: assets/src/__tests__/d3/createMapChart.test.js [hunk 1]
```diff
@@ -0,0 +1,3634 @@
+/* global describe, test, expect */
+import createMapChart from '../../components/d3/createMapChart'
+
+const mapChartData = Object.freeze(
```
[the remark this answers is not in the document]
Comment (soohyunkim) [reply assets/src/__tests__/d3/createMapChart.test.js]: Thanks! Will make a folder under `d3` in `tests`

**Comment by justin0022** [comment]
> I met the error message shown as below when trying to run with `docker exec -it webpack_watcher npm test`:

@Adonais0 could you try again? You'll most likely need to rebuild the docker image. Let us know if it solves this issue.

**Comment by Adonais0** [comment]
Yep, the tests run perfectly for me now. Looks great.

**Pull request merged**